engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/AbstractConfluentKafkaAvroSerializer.scala

```diff
@@ -0,0 +1,95 @@
+package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization
+
+import java.io.{ByteArrayOutputStream, IOException}
+import java.nio.ByteBuffer
+
+import io.confluent.kafka.serializers.{AbstractKafkaAvroSerDe, NonRecordContainer}
+import org.apache.avro.Schema
+import org.apache.avro.generic.{GenericData, GenericDatumWriter, GenericRecord}
+import org.apache.avro.io.EncoderFactory
+import org.apache.avro.reflect.ReflectDatumWriter
+import org.apache.avro.specific.{SpecificDatumWriter, SpecificRecord}
+import org.apache.kafka.common.errors.SerializationException
+
+/**
+  * Abstract confluent serializer class. Serialize algorithm is copy past from AbstractKafkaAvroSerializer.serializeImpl.
+  * Serializer try convert data (in most cases it will be GenericRecord) to indicated schema.
+  *
+  * There is some problem when GenericRecord has different schema then final schema - DatumWriter throws exception,
+  * because data could not support field from schema. When this situation has place wy try to convert data to provided schema by
+  * setting default value when it is possible.
+  *
+  */
+class AbstractConfluentKafkaAvroSerializer extends AbstractKafkaAvroSerDe {
+
+  import scala.collection.JavaConverters._
+
+  protected val encoderFactory: EncoderFactory = EncoderFactory.get
+
+  def serialize(schema: Schema, schemaId: Int, data: Any): Array[Byte] = {
+    if (data == null)
+      null
+    else {
+      try {
+        val out = new ByteArrayOutputStream
+        out.write(AbstractKafkaAvroSerDe.MAGIC_BYTE)
+        out.write(ByteBuffer.allocate(AbstractKafkaAvroSerDe.idSize).putInt(schemaId).array)
+
+        data match {
+          case array: Array[Byte] => out.write(array)
+          case _ =>
+            val encoder = this.encoderFactory.directBinaryEncoder(out, null)
+
+            val value = data match {
```

Comment:
Please extract whole logic of schema evolution to some class like `AvroWriterSchemaEvolutionHandler` or sth similar, with one method `alignRecordToSchema(record, schema)`. We will be able to change implementation and compare if each works the same way.